src/zookeeper.rs

```diff
@@ -524,6 +524,7 @@ impl ZooKeeper {
 
 impl Drop for ZooKeeper {
     fn drop(&mut self) {
+        // TODO: Should track if zookeeper connection has already been closed, and not try to close in drop if it has.
```

Comment:
Added this as the `leader_latch_test_disconnect` test deliberately closes it's `ZooKeeper` instance to assert that leadership is then lost. This results in a non-fatal error being logged from here. Thinking ideally we would keep track of whether the connection had been closed or not, and only close on `drop` if it is still open.